src/enclave/main.cpp

```diff
@@ -56,47 +46,23 @@ extern "C"
       return CreateNodeStatus::NodeAlreadyCreated;
     }
 
-    EnclaveConfig ec = *static_cast<EnclaveConfig*>(enclave_config);
-
     // Setup logger to allow enclave logs to reach the host before node is
     // actually created
     auto circuit = std::make_unique<ringbuffer::Circuit>(
       ringbuffer::BufferDef{
-        ec.to_enclave_buffer_start,
-        ec.to_enclave_buffer_size,
-        ec.to_enclave_buffer_offsets},
+        enclave_config.to_enclave_buffer_start,
+        enclave_config.to_enclave_buffer_size,
+        enclave_config.to_enclave_buffer_offsets},
       ringbuffer::BufferDef{
-        ec.from_enclave_buffer_start,
-        ec.from_enclave_buffer_size,
-        ec.from_enclave_buffer_offsets});
+        enclave_config.from_enclave_buffer_start,
+        enclave_config.from_enclave_buffer_size,
+        enclave_config.from_enclave_buffer_offsets});
     auto basic_writer_factory =
       std::make_unique<ringbuffer::WriterFactory>(*circuit);
     auto writer_factory = std::make_unique<oversized::WriterFactory>(
-      *basic_writer_factory, ec.writer_config);
-
-    // Note: because logger uses ringbuffer, logger can only be initialised once
-    // ringbuffer memory has been verified
-    auto new_logger = std::make_unique<ccf::RingbufferLogger>(*writer_factory);
-    auto* ringbuffer_logger = new_logger.get();
```

Comment:
Small diff with a big semantic impact! All of the "in-enclave" logging, which was nearly everything, used to go via the ringbuffer and now goes directly to stdout! No more happens-before reorderings, but probably _more_ thread races on stdout.